### PR TITLE
Expose predefined security policy resource

### PR DESCRIPTION
**Pull request by annakhm**
This is actually default security policy, named for consistency with
predefined gateway policy (which is not necessarily default).
This resource allows to modify default layer2 or default layer3
security policy.

### CODE REVIEW

File: website/docs/r/policy_predefined_security_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+subcategory: "Policy - Firewall"
+layout: "nsxt"
+page_title: "NSXT: nsxt_policy_predefined_security_policy"
+description: A resource to update Predefined (Default) Security Security Policies.
+---
+
+# nsxt_policy_predefined_security_policy
+
+This resource provides a method to modify default Security Policy and its rules.
+This can be default layer2 policy or default layer2 policy. Maximum one resource
```
Comment (enhaocui) [inline website/docs/r/policy_predefined_security_policy.html.markdown:11]: default layer2 policy or default layer3 policy

File: nsxt/resource_nsxt_policy_predefined_security_policy.go [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+					return err
+				}
+				log.Printf("[DEBUG]: Deleting child rule with id %s", oldRuleID)
+				childRules = append(childRules, childRule)
+
+			}
+		}
+	}
+
+	if d.HasChange("default_rule") {
```
Comment (enhaocui) [inline nsxt/resource_nsxt_policy_predefined_security_policy.go:279]: Maybe also using "else if" here since the two "if"s are mutually exclusive?

**Review by enhaocui — COMMENTED** [review]
Some nits similar to fixed gw policy

File: website/docs/r/policy_predefined_security_policy.html.markdown (hunk 1 above)
Comment (annakhm) [reply website/docs/r/policy_predefined_security_policy.html.markdown]: Thanks for the catch! Done

File: nsxt/resource_nsxt_policy_predefined_security_policy.go (hunk 2 above)
Comment (annakhm) [reply nsxt/resource_nsxt_policy_predefined_security_policy.go:279]: In this resource you can define both rules and default rules.

Comment (enhaocui) [reply nsxt/resource_nsxt_policy_predefined_security_policy.go:279]: Ah I see. Sorry for missing this.

**Review by enhaocui — APPROVED** [review]

**Pull request merged**